### PR TITLE
add clusterlist delete function and code format

### PR DESCRIPTION
**Pull request by lantian0802**
There are not a delete cluster function when click the clusters tab. I add a delete button on the cluster list page and do some code format.

### CODE REVIEW

File: conf/clusters [hunk 1]
```diff
@@ -1,39 +1,4 @@
 {
```
Comment (andypetrella) [inline conf/clusters]: Would it be possible to reintegrate those examples :-D ?

File: public/ipython/tree/js/clusterlist.js [hunk 2]
```diff
@@ -245,6 +254,23 @@ define([
           keyboard_manager: that.keyboard_manager
         });
     });
+
+    delete_button.click(function (e) {
+        var settings = {
+            processData : false,
+            data : "name="+that.data.name,
+            cache : false,
+            type : "GET",
```
Comment (andypetrella) [inline public/ipython/tree/js/clusterlist.js]: would it be better to used `DELETE` here? 
So (below) we can stick with the `cluster` path

File: public/ipython/tree/js/clusterlist.js [hunk 3]
```diff
@@ -245,6 +254,23 @@ define([
...
+            dataType : "json",
+            success : function () {
+                that.load_sessions();
+            },
+            error : utils.log_ajax_error
+        };
+        alert(that.data.name)
+        var url = utils.url_join_encode(that.base_url, 'delCluster');
```
Comment (andypetrella) [inline public/ipython/tree/js/clusterlist.js]: here, I meant, `delCluster` looks weird to me

File: app/controllers/Application.scala [hunk 4]
```diff
@@ -213,6 +213,16 @@ object Application extends Controller {
       }
     }
   }
+  /**
+   * add a spark cluster by json meta
+   */
+  def deleteCluster() = Action.async { request =>
+      println("del a cluster ....")
```
Comment (andypetrella) [inline app/controllers/Application.scala]: consider using this: `Logger.debug("Delete a cluster")`

File: app/controllers/Application.scala [hunk 5]
```diff
@@ -213,6 +213,16 @@ object Application extends Controller {
...
+      val o : JsObject = null;
```
Comment (andypetrella) [inline app/controllers/Application.scala]: it's not used right? Maybe better getting rid of it

File: app/controllers/Application.scala [hunk 6]
```diff
@@ -213,6 +213,16 @@ object Application extends Controller {
...
+      val clusterName = request.queryString.get("name").get.head
+      implicit val ec = kernelSystem.dispatcher
```
Comment (andypetrella) [inline app/controllers/Application.scala]: Maybe adding: `Logger.info("Delete cluster: " + clusterName)` ?

**Comment by andypetrella** [comment]
Really needed PR, thanks a lot @lantian0802 !!!
I just put some minor changes, but besides that :+1: !

File: conf/routes [hunk 7]
```diff
@@ -39,6 +39,7 @@ GET            /nbconvert/:format/*path                  controllers.Application
 
 GET            /clusters                                 controllers.Application.clusters()
 POST           /clusters                                 controllers.Application.addCluster()
+GET            /delCluster                               controllers.Application.deleteCluster()
```
Comment (andypetrella) [inline conf/routes]: So wrt what I say in the `clusterlist.js` file we can keep:
```
DELETE            /clusters                               controllers.Application.deleteCluster()
```

**Comment by lantian0802** [comment]
ok，I will fix the code what you point out.

**Comment by andypetrella** [comment]
@lantian0802 thank you very much :-D

**Pull request merged**